source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/layout/HorizontalFlowLayout.java

```diff
@@ -65,15 +65,29 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
         // need to call super.onMeasure(...) otherwise we'll get some funny behaviour
         super.onMeasure(widthMeasureSpec, heightMeasureSpec);
 
-        final int width = MeasureSpec.getSize(widthMeasureSpec);
+        // Width and height are initially set to be the requested size by the parent
+        int width = MeasureSpec.getSize(widthMeasureSpec);
         int height = MeasureSpec.getSize(heightMeasureSpec);
 
+        final int requiredWidth = measureRequiredWidth(
+            getPaddingLeft(),
+            getPaddingRight());
+
+        if (MeasureSpec.getMode(widthMeasureSpec) == MeasureSpec.UNSPECIFIED) {
+            // set width as required since there's no height restrictions
+            width = requiredWidth;
+        } else if (MeasureSpec.getMode(widthMeasureSpec) == MeasureSpec.AT_MOST
+                && requiredWidth < width) {
+            // set width as required since it's less than the maximum allowed
+            width = requiredWidth;
```

Comment:
since you are setting width to the same value; or seems to be more appropriate. saving the result of MeasureSpec.getMode(widthMeasureSpec) makes sense here.

Reply:
I just made this change and refactored for height as well :)
Let me know what you think